backend/open_webui/routers/users.py

```
@@ -18,48 +26,237 @@
+    if not validate_email_format(form_data.email.lower()):
+        raise HTTPException(
+            status.HTTP_400_BAD_REQUEST, detail=ERROR_MESSAGES.INVALID_EMAIL_FORMAT
+        )
+
+    user = Users.get_user_by_email(form_data.email.lower())
 
+    if user and not user.registration_code:
+        raise HTTPException(400, detail=ERROR_MESSAGES.EMAIL_TAKEN)
+
+    if not user:
+        registration_code = str(random.randint(10**8, 10**9 - 1))
```

Ruff (0.13.1)

Standard pseudo-random generators are not suitable for cryptographic purposes

(S311)